include/cantera/base/Solution.h

```diff
@@ -37,6 +37,12 @@ class Solution : public std::enable_shared_from_this<Solution>
     //! Set the name of this Solution object
     void setName(const std::string& name);
 
+    //! Return the source of this Solution object
+    std::string source() const { return m_source; }
+
+    //! Set the source of this Solution object
+    void setSource(const std::string& source) { m_source = source; }
```

Comment:
It may be simpler, but in many instances PR's eventually end up creating entries in the C++ layer. This can easily be expanded to save source information for objects instantiated in C++ directly.

Reply:
I think it introduces confusion for this value to only be set correctly for the case of a `Solution` instantiated in Python, though. Rather than suggesting that this needs to be populated in those other instances though, I thought having this available only in Python was the simpler fix.

The longer-term option for making this information accessible is that the filename is in the metadata of the `AnyMap` available as `ThermoPhase.input()` for objects created from YAML files. So, once the CTI and XML formats are dead, we can provide this information that way.

Reply:
I think I'm not absolutely certain here: even if it's part of `AnyMap`/YAML, the point here is that neither is saved to HDF - rather, the point of saving `source` in HDF metadata is that things are properly re-created.

Regardless, it is correct that it's only used from the Python interface at the moment, so I'll remove it from the C++ base.